packages/core/src/utils.ts

```diff
@@ -171,3 +171,6 @@ export function isPathParameter(pathSegment: string) {
 export function isNotEmptyObject(obj: any) {
   return !!obj && Object.keys(obj).length > 0;
 }
+
+export const generalizeResponseStatusCode = (code: number | string): string =>
+  `${code}`.replace(/^(\d)\d\d$/, (_, firstDigit) => `${firstDigit}xx`);
```

Comment:
Can you explain this in plain language to me?

Reply:
It translates status codes to general template, e.g. `200` -> `2xx` (see examples in [tests](https://github.com/Redocly/openapi-cli/pull/537/files#diff-ef67a52004653b8e678abb028729aa47bac58239986d6f7afc7317850fe45427R75-R84)). Not sure I named the function correctly though.

Reply:
The issue is there are cases where different headers are common for different response codes -- for example, 200, 201, 202 may have different headers. I think what this needs to do is generalize the other way around.

Reply:
The current approach is: we take a code of the response, and if we find the corresponding header names in the config, we apply the rule. If we don't, we are trying to find if there's a common rule for the code's group (like 2xx, 3xx, &c.). 
I only use the generalization if there's no exact match for the current status code. 
For the example below status code `201` will pick up `[ 'Content-Type' ]`, and `202` will pick up `[ 'x-request-id' ]`: 

```yaml
response-contains-header:
  severity: error
  names:
    2xx: [ 'x-request-id' ]
    201: [ 'Content-Type' ]
```

Do you think it should be a different approach? If yes, could you elaborate?

Reply:
Sorry, I think I misunderstood the direction of the generalization. OpenAPI calls this a range: 2XX is 200-299. I'm still not exactly clear what the method returns. Also, OpenAPI says the field must be a string like '200' and not a number like 200.

This is purely supposed to return the range match. For example, '200' becomes '2XX'. But my question is more like this... let's say there is an API with a responses map with '200', '201', '202', and '204' response codes.

And, let's say there are rules like:

```yaml
response-contains-header:
  severity: error
  names:
    '2xx': [ 'x-request-id' ]
    '201': [ 'Content-Type' ]
```

In this case, all of them must have the `x-request-id`, and in addition to that, the 201 must also have the `Content-Type` header?

What about a name like `convertToStatusCodeRange`?

Reply:
I would rename to `getMatchingStatusCodeRange`

Reply:
> I'm still not exactly clear what the method returns.

It returns a string. If the original code matches the pattern (three digits) - it returns the appropriate code range. Otherwise it returns the input converted to string.

> In this case, all of them must have the x-request-id, and in addition to that, the 201 must also have the Content-Type header?

In my implementation the required headers for a specific code override the required headers for the range. So in your example `201` will only require the `Content-Type` header. I reckon, override is a more obvious option (especially when there is a large set of codes defined) but I might be wrong.